From: Jeff Layton <jlayton@redhat.com>
Date: Mon, 26 Jul 2010 10:53:16 -0400
Subject: [fs] cifs: remove bogus check in NTLM session setup code
Message-id: <1280141596-8210-1-git-send-email-jlayton@redhat.com>
Patchwork-id: 27090
O-Subject: [RHEL5.6 PATCH] BZ#479418: cifs: remove bogus first_time check in
	NTLMv2 session setup code
Bugzilla: 479418
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

This bug appears to be the result of a cut-and-paste mistake from the
NTLMv1 code. The function to generate the MAC key was commented out, but
not the conditional above it. The conditional then ended up causing the
session setup key not to be copied to the buffer unless this was the
first session on the socket, and that made all but the first NTLMv2
session setup fail.

Fix this by removing the conditional and all of the commented clutter
that made it difficult to see.

Tested by Gunther and myself and it resolves the problem.

Cc: Stable <stable@kernel.org>
Reported-by: Gunther Deschner <gdeschne@redhat.com>
Signed-off-by: Jeff Layton <jlayton@redhat.com>

diff --git a/fs/cifs/sess.c b/fs/cifs/sess.c
index abc7fd5..306c9c1 100644
--- a/fs/cifs/sess.c
+++ b/fs/cifs/sess.c
@@ -546,15 +546,7 @@ CIFS_SessSetup(unsigned int xid, struct cifsSesInfo *ses, int first_time,
 
 		/* calculate session key */
 		setup_ntlmv2_rsp(ses, v2_sess_key, nls_cp);
-		if (first_time) /* should this be moved into common code
-				   with similar ntlmv2 path? */
-		/*   cifs_calculate_ntlmv2_mac_key(ses->server->mac_signing_key,
-				response BB FIXME, v2_sess_key); */
-
-		/* copy session key */
-
-	/*	memcpy(bcc_ptr, (char *)ntlm_session_key,LM2_SESS_KEY_SIZE);
-		bcc_ptr += LM2_SESS_KEY_SIZE; */
+		/* FIXME: calculate MAC key */
 		memcpy(bcc_ptr, (char *)v2_sess_key,
 		       sizeof(struct ntlmv2_resp));
 		bcc_ptr += sizeof(struct ntlmv2_resp);
